src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultEditorTemplates.cs

```diff
@@ -378,7 +378,11 @@ public static string TimeInputTemplate(IHtmlHelper htmlHelper)
 
         public static string NumberInputTemplate(IHtmlHelper htmlHelper)
         {
-            return GenerateTextBox(htmlHelper, inputType: "number");
+            var htmlAttributes =
+                CreateHtmlAttributes(htmlHelper, className: "text-box single-line", inputType: "number");
+            htmlAttributes["step"] = "any";
```

Comment:
I thought this was needed for only decimal-enabled types: `float`, `double`, and `decimal`. For integral types there's no step needed, right?